tools/pylib/boututils/showdata.py

```diff
@@ -585,19 +641,25 @@ def init():
 
     #If movie is not passed as a string assign the default filename
     if (movie==1):
-        movie='animation'
+        movie='animation.mp4'
 
     # Save movie with given or default name
     if ((isinstance(movie,str)==1)):
-        try:
-            anim.save(movie+'.mp4',writer = FFwriter, fps=fps, dpi=dpi, extra_args=['-vcodec', 'libx264'])
-        except Exception:
-        #Try specifying writer by string if ffmpeg not found
+        movietype = movie.split('.')[-1]
+        if movietype == 'mp4':
+            try:
+                anim.save(movie,writer = FFwriter, fps=fps, dpi=dpi, extra_args=['-vcodec', 'libx264'])
+            except Exception:
+            #Try specifying writer by string if ffmpeg not found
                 try:
-                    anim.save(movie+'.mp4',writer = 'ffmpeg', fps=fps, dpi=dpi, extra_args=['-vcodec', 'libx264'])
+                    anim.save(movie,writer = 'ffmpeg', fps=fps, dpi=dpi, extra_args=['-vcodec', 'libx264'])
                 except Exception:
                      print('Save failed: Check ffmpeg path')
                      raise
+        elif movietype == 'gif':
+            anim.save(movie+'.gif',writer = 'imagemagick', fps=fps, dpi=dpi, extra_args=['-vcodec', 'libx264'])
```

Comment:
`extra_args` can be dropped here -- `vcodec` is an ffmpeg option, and `libx264` is the mpeg-4 codec